src/app/reporting/Engine.cpp

```diff
@@ -421,6 +421,7 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
     const uint32_t kReservedSizeForEventReportIBs = 3; // type, tag, end of container
 
     VerifyOrExit(apReadHandler != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(apReadHandler->GetSession() != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
```

Comment:
What's motivating adding this check here?

Reply:
Previously, we made a copy of the SessionDescriptor, and the report generation will pass and the send report will fail.

To reduce the size of ReadHandler, I removed the SessionDescriptor, thus we need a valid Session or it will crash (dereference nullptr).